### PR TITLE
[OSDOCS-11001]: Moving HCP bare metal content to OCP repo

### PR DESCRIPTION
**Pull request by lahinson**
Version(s): 4.17

Issue: https://issues.redhat.com/browse/OSDOCS-11005

Link to docs preview:
- "Deploying hosted control planes on bare metal": https://80904--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-deploy/hcp-deploy-bm.html
- "Managing hosted control planes on bare metal": https://80904--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-manage/hcp-manage-bm.html
- https://80904--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-import.html

QE review:
- [x] QE has approved this change.

NOTE FOR PEER REVIEWERS: This PR moves the bare metal HCP content from the RHACM docs to the OCP docs. The content has not been changed since it was SME- and peer-reviewed by the RHACM team. This PR simply moves the content and applies OCP docs-specific formatting.

### CODE REVIEW

File: modules/hcp-bm-hc.adoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+----
++
+Replace `<hosted_cluster_namespace>` with your hosted cluster namespace name, for example, `clusters`. Replace `<hosted_cluster_name>` with your hosted cluster name.
+
+. Verify that you have a default storage class configured for your cluster. Otherwise, you might see pending PVCs. Run the following command:
++
+[source,terminal]
+----
+$ hcp create cluster agent \
+    --name=<hosted_cluster_name> \ <1>
```
Comment (kcarmichael08) [inline modules/hcp-bm-hc.adoc:35]: Callouts in multi-line commands need to be commented out per [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#callouts-in-code-blocks).

Example:

```
[source,terminal]
----
$ oc autoscale <object_type>/<name> \// <1>
```

File: modules/hcp-import-disable.adoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// Module included in the following assemblies:
+//
+// * hosted_control_planes/hcp-import.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="hcp-import-disable_{context}"]
+= Disabling the automatic import of hosted clusters into {mce-short}
+
+Hosted clusters are automatically imported into {mce-short} after the control plane becomes available, you can disable the automatic import of hosted clusters.
```
Comment (kcarmichael08) [inline modules/hcp-import-disable.adoc:9]: run-on sentence?

File: modules/hcp-import-manual-aws.adoc [hunk 3]
```diff
@@ -0,0 +1,89 @@
+// Module included in the following assemblies:
+//
+// * hosted_control_planes/hcp-import.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="hcp-import-manual-aws_{context}"]
+= Manually importing a hosted control plane cluster on AWS
+
+You can also import a hosted control plane cluster on AWS with the command line interface by completing the following steps.
```
Comment (kcarmichael08) [inline modules/hcp-import-manual-aws.adoc:9]: I think AWS should be spelled out at first use.

**Review by kcarmichael08 — COMMENTED** [review]
Hey, this looks really great overall! I marked some issues per the OpenShift guidelines, and had a couple of questions.

**Comment by kcarmichael08** [comment]
(I didn't closely review the body of the text since it's already gone through peer review but a couple of things jumped out at me so I added comments.)

File: hosted_control_planes/hcp-deploy/hcp-deploy-bm.adoc [hunk 4]
```diff
@@ -5,3 +5,70 @@ include::_attributes/common-attributes.adoc[]
...
+
+The hosted cluster is automatically imported as a managed cluster. If you want to disable this automatic import feature, see _Disabling the automatic import of hosted clusters into {mce-short}_.
+
+include::modules/hcp-bm-prepare.adoc[leveloffset=+1]
+include::modules/hcp-bm-prereqs.adoc[leveloffset=+2]
+
+[role="_additional-resources"]
+.Additional resources
+
+* For more information about the `local-cluster`, see link:https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.11/html/clusters/cluster_mce_overview#advanced-config-engine[Advanced configuration].
```
[the remark this answers is not in the document]
Comment (lahinson) [reply hosted_control_planes/hcp-deploy/hcp-deploy-bm.adoc:30]: Good catch! I'll update the links to show only the text inside the brackets.

File: modules/hcp-bm-hc-console.adoc [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+To create a hosted cluster by using the console, complete the following steps.
+
+.Procedure
+
+. Open the {product-title} web console and log in by entering your administrator credentials. For instructions to open the console, see _Accessing the web console_.
+
+. In the console header, ensure that **All Clusters** is selected.
+
+. Click **Infrastructure > Clusters**.
```
[the remark this answers is not in the document]
Comment (lahinson) [reply modules/hcp-bm-hc-console.adoc:17]: Good catch! Will fix.

File: modules/hcp-bm-prereqs.adoc [hunk 6]
```diff
@@ -0,0 +1,22 @@
+// Module included in the following assemblies:
+//
+// * hosted_control_planes/hcp-deploy/hcp-deploy-bm.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="hcp-bm-prereqs_{context}"]
+= Prerequisites to configure a management cluster
+
+* You need the {mce} 2.2 and later installed on an {product-title} cluster. You can install {mce-short} without as an Operator from the {product-title} OperatorHub.
```
[the remark this answers is not in the document]
Comment (lahinson) [reply modules/hcp-bm-prereqs.adoc:9]: That's a typo. I'll fix it. Thanks!

**Comment by lahinson** [comment]
Thanks so much for your review, @kcarmichael08! Good points all around. I'll incorporate your suggestions now :)

**Pull request merged**